ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -47,6 +48,16 @@ import { getSeriesColor } from './utils/palette-gen';
 
 export type TimeSeriesChartProps = PanelProps<TimeSeriesChartOptions>;
 
+// Using an "ALL" value to handle the case on first loading the chart where we
+// want to select all, but do not want all of the legend items to be visually highlighted.
+// This helps us differentiate those cases more clearly instead of inferring it
+// based on the state of the data. This also helps us avoid some coding
+// complexity around initializing a full record for the initial load that would
+// currently require significantly more refactoring of this component.
+// TODO: simplify this if we switch the list-based legend UI to use checkboxes,
+// where we *would* want to visually select all items in this case.
+type SelectedSeriesState = Record<string, boolean> | 'ALL';
```

Comment:
Swapping from an array to an object because:
- It'll line up better with the state needed for table legend 
- It should be more performant for lookup operations.